### ISSUE TITLE
Improve state of docs

### ISSUE THREAD

**Issue by pbrehmer** [mfherbst/ReducedBasis.jl#36]
There are a few things, that could be improved and added:
- the axes labels of the plots are cut off
- the DMRG example output should be improved by fixing the number of snapshots etc. (produces unstable plots at the moment)
- add an example showcasing truncation, simulation continuation and some other "basis analysis" features
- maybe use `Literate.jl` to avoid code duplication with respect to examples/* (see also #15)
- implement Stefan's feedback
- go through all Docstrings again and see what is really necessary or could be improved

**Comment by mfherbst** [comment]
I also have some feedback we could talk about. Most importantly I feel you are assuming people read the docs from front to back, but spoiler: They won't. Make sure each docs page is self-contained or directly references a page where some aspect is explained. In particular don't hide code. Some people will copy and past and walk away frustrated if it does not just work.

**Comment by pbrehmer** [comment]
Yes that's a very good point. The point of hiding code was to show that only certain steps in the offline phase have to be changed to customize the solver etc., while all of the online procedure stays the same.

Now with `Literate` I won't hide any code and just remark on the bits that previous examples have already covered, to highlight this.

**Comment by pbrehmer** [comment]
PR #37 fixed most of these things. What's left to do is:

- improving and cleaning up docstrings

And additionally to get the docs to a good "preliminary state":

- implement small improvements discussed earlier today (rename `result` to `rbres`, remove unnecessary kwargs, ...)
- add introduction page with basic RB theory and theory surrounding the application to quantum many-body systems
